spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/TracingProperties.java

```diff
@@ -166,29 +168,54 @@ public void setFields(List<String> fields) {
 	public static class Propagation {
 
 		/**
-		 * Tracing context propagation type.
+		 * Tracing context propagation types produced by the application.
 		 */
-		private PropagationType type = PropagationType.W3C;
+		private List<PropagationType> type = List.of(PropagationType.W3C);
 
-		public PropagationType getType() {
+		public List<PropagationType> getType() {
 			return this.type;
 		}
 
-		public void setType(PropagationType type) {
+		public void setType(List<PropagationType> type) {
 			this.type = type;
 		}
 
 		enum PropagationType {
 
 			/**
-			 * B3 propagation type.
+			 * <a href="https://github.com/openzipkin/b3-propagation#single-header">B3
+			 * single header</a> propagation.
 			 */
-			B3,
+			B3(1),
 
 			/**
-			 * W3C propagation type.
+			 * <a href="https://github.com/openzipkin/b3-propagation#multiple-headers">B3
+			 * multiple headers</a> propagation.
 			 */
-			W3C
+			B3_MULTI(2),
+
+			/**
+			 * W3C propagation.
+			 */
+			W3C(0);
+
+			/**
+			 * Order of the enum value. The {@link #orderedValues()} method returns the
+			 * enum values in ascending order.
+			 */
+			private final int order;
+
+			PropagationType(int order) {
+				this.order = order;
+			}
+
+			/**
+			 * Returns the ordered values.
+			 * @return the ordered values
+			 */
+			static List<PropagationType> orderedValues() {
```

Comment:
Enums are already ordered. Could we change the order in which the values are declared instead?

Reply:
Sure! I preferred the explicit ordering here, but that's no strong opinion on my side.

Reply:
I think I would go without having `order` on this enum too and add a line to the javadocs that the order these instances are defined matters.

Reply:
Changed it.